Source/JavaScriptCore/runtime/MachineContext.h

```diff
@@ -800,49 +812,37 @@ inline void*& llintInstructionPointer(mcontext_t& machineContext)
     return llintInstructionPointer(machineContext->__ss);
 #elif OS(FREEBSD)
 
-#if CPU(X86)
-    return reinterpret_cast<void*&>((uintptr_t&) machineContext.mc_esi);
-#elif CPU(X86_64)
+#if CPU(X86_64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.mc_r8);
 #elif CPU(ARM)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_R8]);
 #elif CPU(ARM64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.mc_gpregs.gp_x[4]);
-#elif CPU(MIPS)
-    return reinterpret_cast<void*&>((uintptr_t&) machineContext.mc_regs[12]);
 #else
 #error Unknown Architecture
 #endif
 
 #elif OS(NETBSD)
 
-#if CPU(X86)
-    return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_ESI]);
-#elif CPU(X86_64)
+#if CPU(X86_64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_R8]);
 #elif CPU(ARM)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_R8]);
 #elif CPU(ARM64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_X4]);
-#elif CPU(MIPS)
-    return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_T4]);
 #else
 #error Unknown Architecture
 #endif
 
 #elif OS(FUCHSIA) || OS(LINUX) || OS(HURD)
 
 // The following sequence depends on glibc's sys/ucontext.h.
-#if CPU(X86)
-    return reinterpret_cast<void*&>((uintptr_t&) machineContext.gregs[REG_ESI]);
-#elif CPU(X86_64)
+#if CPU(X86_64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.gregs[REG_R8]);
 #elif CPU(ARM)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.arm_r8);
 #elif CPU(ARM64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.regs[4]);
-#elif CPU(MIPS)
-    return reinterpret_cast<void*&>((uintptr_t&) machineContext.gregs[12]);
```

Comment:
Do people still occasionally do drive by MIPS fixes? Maybe we should just leave this for now? We can always do a full `unifdef !CPU(MIPS)` another time.

Reply:
MachineContext is optional and we do not need to have that for MIPS since it is not having JIT.